testdata/p4_16_samples/ebpf/action_call_ebpf.p4

```diff
@@ -25,7 +25,7 @@ parser prs(packet_in p, out Headers_t headers) {
     }
 }
 
-control pipe(inout Headers_t headers, out bool pass) {
+control pipe_action_call_ebpf(inout Headers_t headers, out bool pass) {
```

Comment:
Why are you renaming all these controls?

Reply:
ebpf maps are globally unique, if the table/control names are identical parallel execution leads to conflict. I could also rename the table instead of the control function. This an issue specific to the way we run tests in parallel

Reply:
That is actually a problem. Ideally we would prepend the folder name or the file name to the table name, but I don't know if that's possible the way we do things now. We should talk about how to do this properly.

Reply:
Using namespaces this problem can be avoided entirely. All maps are local to the specific namespace (or process). However, I am encountering a nasty issue, the map files do not persist across namespace calls. For example, if I call 

ip netns exec "namespace" tc filter add "ebpfprogram" 
which exports the tables, then subsequently call
ip netns exec runtime
 which accesses the tables, I get an error that the tables do not exist. The maps are instantly garbage collected once the ip netns call finishes. On the other hand, the program remains attached to the interface. 

This issue can be mitigated by running:
ip netns exec "namespace" bash -c "tc filter add "ebpfprogram";  ip netns exec runtime" 
as a single command. This is obviously not ideal. 

Not sure what to do about this, I am also not sure if this is working as intended.

Reply:
Actually, it is working as intended, running with a prefixed "nohup" seems to do the trick.